From ddd5099c6dd21289d3f5b3ee78863a03b831c32d Mon Sep 17 00:00:00 2001
From: David Daney <ddaney@caviumnetworks.com>
Date: Fri, 10 Jun 2011 16:53:45 -0700
Subject: [PATCH 090/974] MIPS: Octeon: Don't restore $ra in SAVE_SOME

The $ra value is never used, so we can save an instruction by not
restoring it.

Signed-off-by: David Daney <ddaney@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/include/asm/stackframe.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/mips/include/asm/stackframe.h b/arch/mips/include/asm/stackframe.h
index 7c67f6b..67642e0 100644
--- a/arch/mips/include/asm/stackframe.h
+++ b/arch/mips/include/asm/stackframe.h
@@ -226,8 +226,6 @@
 		    corrupt it */
 		jal	octeon_mult_save
 		LONG_L	v1, 0($28)  /* Load the current pointer */
-			 /* Restore $31(ra) that was changed by the jal */
-		LONG_L	ra, PT_R31(sp)
 		pref	0, 0(v1)    /* Prefetch the current thread */
 #endif
 		.set	pop
-- 
2.6.2

